modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/dml/UpdatePlan.java

```diff
@@ -646,20 +646,21 @@ private AbstractIterator(ConnectionManager connMgr, QueryCursor<List<?>> cur, Up
 
         /** {@inheritDoc} */
         @Override public void beforeDetach() {
-            ThreadLocalObjectPool<H2ConnectionWrapper>.Reusable conn0 = conn = connMgr.detachThreadConnection();
+            // TODO: wrong!
```

Comment:
Remove TODO or create a ticket.

Reply:
AbstractIterator is refactored. It doesn't need in connection now.
The cursor holds it.